### PR TITLE
JP-2658: Update WCS calculations for the GUIDING modes

### LINKED ISSUE #7282: FGS WCS CRPIX values incorrectly calculated
_Issue [JP-2658](https://jira.stsci.edu/browse/JP-2658) was created on JIRA by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham):_

FGS CRPIX values are being incorrectly pulled from the SIAF.

The values should be set, as described in the 5.3.4 of Guide Star Pipeline Data Product Design JWST-STScI-004800, SM-12.

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Resolves [JP-2658](https://jira.stsci.edu/browse/JP-2658)

Closes #7282 

This PR addresses an issue where the appropriate algorithms to calculation the CRPIX* values of the WCS for the various GUIDING exposures was not used.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

Note: This PR supersedes #6975

### CODE REVIEW

File: jwst/lib/tests/build_engdb.ipynb [hunk 1]
```diff
@@ -66,6 +67,72 @@
...
+    "#date_end = Time('2022-05-23T00:37:0')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "engdb = EngdbMast(token='8fb3131048d448f9997f5d76df74e9b3')"
```
Comment (drlaw1558) [inline jwst/lib/tests/build_engdb.ipynb:124]: Should this MAST token be included here in a public notebook?

Comment (stscieisenhamer) [reply jwst/lib/tests/build_engdb.ipynb:124]: Good catch. Oh probably not, though it is also stale. Will remove. 😱

File: CHANGES.rst [hunk 2]
```diff
@@ -184,7 +184,13 @@ residual_fringe
 
 - Use scipy.interpolate.BSpline instead of astropy.modeling.Spline1D in
   residual_fringe fitting utils [#7764]
-  
+
+
+set_telescope_pointing
+----------------------
+
+-  Update WCS calculations for the GUIDING modes [#7889]
```
Comment (hbushouse) [inline CHANGES.rst:192]: Can you add a little detail describing what the updates were?

**Review by hbushouse — APPROVED** [review]
Just one trivial comment.

**Comment by hbushouse** [comment]
Has there been another regtest attempt?

**Review by hbushouse — APPROVED** [review]

**Review by drlaw1558 — APPROVED** [review]
Empirically speaking, looks good to me.  Rerunning stp.add_wcs on an FGS ACQ image I now get a WCS embedded into the FITS file that lines the star up with the nominal target coordinates to the expected few-pixel accuracy.

**Comment by stscieisenhamer** [comment]
[Regression try #56244](https://plwishmaster.stsci.edu:8081/job/RT/job/eisenhamer-dev/349/) using a modified JenkinsfilesRT_dev to get around the repo check.

One failure relevant to the modified code was a, you guessed it, segmentation violation. [Another run](https://plwishmaster.stsci.edu:8081/job/RT/job/eisenhamer-dev/350/) is being done on just the one test to see if it was an isolated/loaded server incident. 

There are also a number of errors with respect to NIRSpec FS Spec3, which is not related, but others will need to chime in on whether this is known or not.

**Comment by hbushouse** [comment]
Yes, the NIRSpec FS spec3 failures are due to the merger of a PR of mine yesterday. Those tests have now been updated, which includes not just updated data files on Artifactory, but also updated test code in jwst/regtest. So if you rebase the PR branch to pull in that updated test module, it should work now.

And given that this change only affects STP, for which we only have like 1 module of tests, is it even really necessary to run the whole regtest suite?  Just running that one module would be sufficient (IMHO).

**Comment by hbushouse** [comment]
Given that: 1) the regtest servers are hosed, 2) the set_telescope_pointing module only affects 1 regtest module, 3) Jonathan has successfully run that 1 test module locally, and 4) the CI tests look good,  I'm going to merge without record of a full regtest run.

**Comment by hbushouse** [comment]
@nden Are you OK with the latest state of the code here?

**Comment by stscieisenhamer** [comment]
The one CI failure is a Numpy 2.0 issue. That will be dealt with in the next release. Merging to get into the regression run.

**Pull request merged**